modern/gcc/Dockerfile

```diff
@@ -0,0 +1,74 @@
+ARG DOCKER_USERNAME
+
+FROM ubuntu:xenial as builder
+
+ARG GCC_VERSION ARTIFACTORY_REPOSITORY LIBSTDCPP_VERSION
+
+LABEL maintainer="Conan.io <info@conan.io>"
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends --no-install-suggests -o=Dpkg::Use-Pty=0 \
+       g++-multilib \
+       gcc \
+       curl \
+       wget \
+       zlib1g \
+       ca-certificates \
+       autoconf-archive \
+       binutils \
+       libgmp-dev \
+       libmpfr-dev \
+       libmpc-dev \
+       nasm \
+       dh-autoreconf \
+       libffi-dev \
+       libssl-dev \
+       pkg-config \
+    && wget -q --no-check-certificate http://mirrors.concertpass.com/gcc/releases/gcc-${GCC_VERSION}/gcc-${GCC_VERSION}.tar.xz \
+    && tar Jxf gcc-${GCC_VERSION}.tar.xz \
+    && cd gcc-${GCC_VERSION} \
+    && ./configure --build=x86_64-linux-gnu \
```

Comment:
may we download pre-built compiler if available from artifactory generic repository here?
I don't want to waste 1 hour * number of compilers each time I modify the image
(e.g. bump `6.0.29` -> `6.0.30`)
I think rebuild of the compiler is really rarely (RARELY) required, only if we modify e.g. configure flags for the compiler itself.

Reply:
I prefer keep the recipe "how to build from sources", as an enterprise, I would not like to download a magic package with binaries. The builder stage is isolated from others, is not something to be rebuilt every time, you build once, cache it and only rebuild the Deploy step.